File: 530/minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution():
    def getMinimumDifference(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def dfs(node):
            if node == None:
                return
            res.append(node.val)
            if node.left:
                dfs(node.left)
            if node.right:
                dfs(node.right)

        res = []
        dfs(root)
        res = sorted(res)

        ans = 99999999
        for i in range(len(res)):
            if (i):
                ans = min(ans, res[i] - res[i - 1])

        return ans

File: 530/minimum-absolute-difference-in-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var getMinimumDifference = function (root) {
    let ans = [];
    function dfs(root) {
        if (!root) {
            return;
        } else {
            ans.push(root.val);
        }
        if (!root.left)
            dfs(root.left);
        if (!root.right)
            dfs(root.right);
    }

    dfs(root);
    ans.sort((a, b) => a - b);
    let minDiff = Infinity;
    for (let i = 1, len = ans.length; i < len; i++) {
        minDiff = Math.min(minDiff, ans[i] - ans[i - 1]);
    }
    return minDiff;
};